src/utils.js

```diff
@@ -107,7 +102,15 @@ export async function sign(params = { }) {
 
   const privateKeyBytes = base64url.baseDecode(privateJwk.d);
 
-  const signatureBytes = await signer.sign(messageBytes, privateKeyBytes, signerOpts);
+  // sign the actual payload
+  let signatureBytes;
+  if (privateJwk.crv === 'Ed25519') {
+    signatureBytes = await ed25519.signAsync(messageBytes, privateKeyBytes);
+  }
+  else if (privateJwk.crv === 'secp256k1') {
+    const signature = await secp256k1.signAsync(messageBytes, privateKeyBytes);
```

Comment:
is this intentional?

Reply:
The fact that it looks different to the ED25519 path? Yes, one of the v2 breaking changes.